src/sortStudents.ts

```diff
@@ -1,16 +1,61 @@
 
 export interface Student {
-  // describe Student interface
+  name: string;
+  surname: string;
+  age: number;
+  married: boolean;
+  grades: number[];
 }
 
 export enum SortType {
-  // describe SortType enum
+  Name = 'name',
+  Surname = 'surname',
+  Age = 'age',
+  Married = 'married',
+  AverageGrade = 'averageGrade',
 }
 
-// create SortOrder type
-export type SortOrder;
+export type SortOrder = 'asc' | 'desc';
 
+function getAverageGrade(grades: number[]): number {
+  return grades.reduce((sum, prev) => sum + prev, 0) / grades.length;
+}
+
+export function sortStudents(
+  students: Student [],
+  sortBy: SortType,
+  order: SortOrder,
+): Student [] {
+  const studentsClone = [...students];
+
+  return studentsClone.sort((studentA, studentB) => {
+    switch (sortBy) {
+      case SortType.Name:
+
+        return order === 'asc'
+          ? studentA.name.localeCompare(studentB.name)
+          : studentB.name.localeCompare(studentA.name);
+
+      case SortType.Surname:
+
+        return order === 'asc'
+          ? studentA.surname.localeCompare(studentB.surname)
+          : studentB.surname.localeCompare(studentA.surname);
```

Comment:
These two have 100% identical logic. Switch allows combining several cases when put without brake command
It can be said about next 2 cases as well
```suggestion
      case SortType.Name:
      case SortType.Surname:

        return order === 'asc'
          ? studentA.surname.localeCompare(studentB.surname)
          : studentB.surname.localeCompare(studentA.surname);
```

Reply:
I'm not sure this is a good point

Reply:
This is a good point, but to do this you need to use computed keys
For example
`studentA[sortBy].localCompare(studentB[sortBy])`